programs/degen-pools/src/pools.rs

```diff
@@ -87,6 +89,45 @@ pub fn set_winning_option(ctx: Context<UpdatePool>, winning_option: Pubkey) -> R
     Ok(())
 }
 
+pub fn fund_pool<'c: 'info, 'info>(
+    ctx: Context<'_, '_, 'c, 'info, FundPool<'info>>,
+    value: u64,
+) -> Result<()> {
+    let pool_account = &mut ctx.accounts.pool_account;
+    if pool_account.is_paused {
+        return err!(CustomError::PoolStateIncompatible);
+    }
+
+    if ctx.remaining_accounts.len() == 0 {
+        return err!(CustomError::NoPoolOptionAccounts);
+    }
+
+    let remaining_accounts = ctx.remaining_accounts;
+    let value_per_option = value / remaining_accounts.len() as u64;
+    for account in remaining_accounts {
+        let mut pool_option = match Account::<PoolOption>::try_from(account) {
+            Ok(pool_option) => pool_option,
+            Err(_) => return err!(CustomError::InvalidPoolOptionAccount),
```

Comment:
We should not just check that it is a valid PoolOption but also check that it belongs to the pool that was passed in the instruction. If we follow [this suggestion](https://github.com/Degen-Markets/degen-markets-solana/pull/12/files#r1829242517), `enter_pool` will do the check for us

Reply:
Will pursue